packages/react-core/src/demos/PasswordGenerator.md

```diff
@@ -0,0 +1,91 @@
+---
+id: Password generator
+section: demos
+---
+
+import RedoIcon from '@patternfly/react-icons/dist/esm/icons/redo-icon';
+import EyeIcon from '@patternfly/react-icons/dist/esm/icons/eye-icon';
+import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
+
+## Examples
+
+### Provide a generated password
+
+```js
+import React from 'react';
+import {
+  Form,
+  FormGroup,
+  InputGroup,
+  TextInput,
+  Button,
+  Popover,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  MenuItemAction
+} from '@patternfly/react-core';
+import RedoIcon from '@patternfly/react-icons/dist/esm/icons/redo-icon';
+import EyeIcon from '@patternfly/react-icons/dist/esm/icons/eye-icon';
+import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
+
+PasswordGenerator = () => (
+  <Form>
+    <FormGroup
+      label="Password"
+      isRequired
+      fieldId="password-generator-demo--initial-password"
+      labelIcon={
+        <Popover bodyContent={<div>Generates a password</div>}>
+          <button
+            type="button"
+            aria-label="More info"
+            onClick={e => e.preventDefault()}
+            className="pf-c-form__group-label-help"
+          >
+            <HelpIcon noVerticalAlign />
+          </button>
+        </Popover>
+      }
+    >
+      <InputGroup>
+        <TextInput
+          isRequired
+          type="password"
+          id="password-generator-demo--initial-password"
+          name="password-generator-demo--initial-password"
+          aria-label="Password input"
+          placeholder="Password"
+          iconVariant="fa-eye"
+        />
+        <Button variant="control" aria-label="search button for search input">
+          <EyeIcon />
+        </Button>
+      </InputGroup>
+      <Menu isPlain>
```

Comment:
So the HTML demo that this is modeled after is meant to have the menu appear as a sort of dropdown when the user puts focus into the password input.

And usually the demos we build in react are supposed to have all interactions implemented. So the 'refresh' button should suggest a new password, and clicking the suggestion should have it auto populate the password field.

The demo should also be fully keyboard accessible - so I'd look at an existing demo - like the [search input with autocomplete](ct-demos#search-with-autocomplete) demo as an example for how using arrow keys, tab, escape, enter, etc should allow the user to interact with the suggested password

Reply:
Also the show/hide (eye icon) button should toggle the text input type, showing and hiding the password in the input field. We have a login demo that shows that interaction - https://www.patternfly.org/v4/components/login-page/react/showhide-password/